File: lab-2.1/c#/Segment.cs
using System;

public class Segment : Line
{
    public Segment() : base() { }
    public Segment(double x1, double y1, double x2, double y2) : base(x1, y1, x2, y2) { }
    public Segment(Segment other) : base(other) { }

    public double CalculateAngleWithOY()
    {
        double dx = x2 - x1;
        double dy = y2 - y1;
        return Math.Atan2(dx, dy) * (180 / Math.PI);
    }
}

File: lab-2.1/c++/Segment.cpp
#include "Segment.h"
#include <cmath>

Segment::Segment() : Line() {}

Segment::Segment(double x1, double y1, double x2, double y2)
    : Line(x1, y1, x2, y2) {
}

Segment::Segment(const Segment& other) : Line(other) {}

double Segment::calculateAngleWithOY() const {
    double dx = x2 - x1;
    double dy = y2 - y1;
    return atan2(dx, dy) * 180 / 3.14;
}